packages/ui/primitives/signature-pad/signature-pad.tsx

```diff
@@ -183,8 +184,11 @@ export const SignaturePad = ({
     // Clear the canvas
     if ($el.current) {
```

Comment:
Another thing I observed is that when you undo the first change to an already existing signature, the signature is not updated. See the video to better understand what I mean.

https://github.com/documenso/documenso/assets/25515812/281b776d-8366-4116-8bb9-0df5d6f1cfdb

I tried calling the `onChange` function after clearing the canvas to update the signature URL and it seems to work.

```js
if ($el.current) {
      const ctx = $el.current.getContext('2d');
      const { width, height } = $el.current;
      ctx?.clearRect(0, 0, width, height);
      if (typeof defaultValue === 'string' && defaultImageRef.current) {
        ctx?.putImageData(defaultImageRef.current, 0, 0);
      }
      newLines.forEach((line) => {
        const pathData = new Path2D(getSvgPathFromStroke(getStroke(line, perfectFreehandOptions)));
        ctx?.fill(pathData);
      });

      /*
         here
      */
      onChange?.($el.current.toDataURL());
    }
```

Please double check and see if it works and/or if there is a better way of doing it.

Here's how it works with the new addition.

https://github.com/documenso/documenso/assets/25515812/10a95739-e0b5-4889-894b-5d57df96c136

Reply:
@catalinpit Thanks for pointing out. This makes sense, I've added an onChange handler. This is a good method to work with it :)

Reply:
@catalinpit Let me know if the changes are fine :)